### PR TITLE
ISSUE-133 updated config field for host,port,context

### PR DESCRIPTION
**Pull request by cooljavadev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,9 +239,9 @@ You just need the below `hello_world_status_ok_assertions.json`.
 
 the `github_host.properties` looks as below:
 ```
-restful.application.endpoint.host=https://api.github.com
-restful.application.endpoint.port=443
-restful.application.endpoint.context=
+web.application.endpoint.host=https://api.github.com
```
Comment (authorjapps) [inline README.md]: 👍

**Comment by authorjapps** [comment]
Looks great @cooljavadev !

Can you please make this backward compatible for existing users. It should be easy one- _some pointers below:_

There is a property loader method in the DI i.e. `org.jsmart.zerocode.core.di.ApplicationMainModule#getProperties` 
You can add a bit of logic to check if `web...` is null, then go for `restful...`, something like below-
```
/// backward compatibility ///
  if(properties.get(WEB_APPLICATION_ENDPOINT_HOST) == null){
     properties.setProperty(WEB_APPLICATION_ENDPOINT_HOST, (String)properties.get(RESTFUL_APPLICATION_ENDPOINT_HOST));
  }
  // repeat for all three properties

  return properties;
}
```
where(you can declare then as `static, final` **constants** )-
>     String WEB_APPLICATION_ENDPOINT_HOST = "web.application.endpoint.host";
>     String RESTFUL_APPLICATION_ENDPOINT_HOST = "restful.application.endpoint.host";

- Just repeat the above for all `three(3)` properties keys

File: README.md (hunk 1 above)
Comment (cooljavadev) [reply README.md]: code changes updated for backword compatibility

**Comment by authorjapps** [comment]
Yes, please @cooljavadev , @aghorpade , please add a section to the [TOC](https://github.com/authorjapps/zerocode/blob/master/README.md#table-of-contents--) and bit of explanation something like this-
- From the release 1.2.8 onwards we will be allowing `web.` and deprecating `restful.`
- We will take away support for `restful.` in future releases
- 1.2.8 will work for both as we have made this backward compatible.

File: src/main/java/org/jsmart/zerocode/core/di/ApplicationMainModule.java [hunk 2]
```diff
@@ -58,7 +65,16 @@ public Properties getProperties(String host) {
         final Properties properties = new Properties();
         try {
             properties.load(getClass().getClassLoader().getResourceAsStream(host));
-
+            /// below code is for backward compatibility, remove this after few releases ///
+            if(properties.get(WEB_APPLICATION_ENDPOINT_HOST) == null){
```
Comment (authorjapps) [inline src/main/java/org/jsmart/zerocode/core/di/ApplicationMainModule.java]: A test please proving both works. Again this shd be easy enough, while you did a replace all, keep one of the place as `restful...`, then the test should not fail.
> If you have already taken care, then you can ignore this comment.

**Comment by cooljavadev** [comment]
> Yes, please @cooljavadev , @aghorpade , please add a section to the [TOC](https://github.com/authorjapps/zerocode/blob/master/README.md#table-of-contents--) and bit of explanation something like this-
> 
> * From the release 1.2.8 onwards we will be allowing `web.` and deprecating `restful.`
> * We will take away support for `restful.` in future releases
> * 1.2.8 will work for both as we have made this backward compatible.

Changes done. Review once

File: src/main/java/org/jsmart/zerocode/core/di/ApplicationMainModule.java (hunk 2 above)
Comment (cooljavadev) [reply src/main/java/org/jsmart/zerocode/core/di/ApplicationMainModule.java]: @authorjapps  added new test case with new property file ,so we can easily remove these two files after backward compatibility feature is removed

**Comment by authorjapps** [comment]
@cooljavadev , Couple of things if you could take care after the PR merge please 🙏.
- Update the example in the HelloWorld git repo project (same procedure, raise a PR once done)
- It's under `src/test/java/org/jsmart/zerocode/testhelp/tests/helloworld`
> https://github.com/authorjapps/zerocode-hello-world/tree/master/src/test/java/org/jsmart/zerocode/testhelp/tests/helloworld

> JustHelloWorldTest.java

- Just update the keys in the "github_host.properties", that should do for now 👍

**Pull request merged**